kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -380,6 +380,8 @@ def create_kcfg_explore() -> KCFGExplore:
             ready = topological_sorter.get_ready()
             _LOGGER.info(f'Discharging proof obligations: {ready}')
             curr_claim_list = [all_claim_jobs_by_label[label] for label in ready]
+            for claim in curr_claim_list:
+                claim.up_to_date(digest_file)
```

Comment:
Eliminating potential data races coming from line 385 below, since `up_to_date` both reads and writes to digest file.

Reply:
Nice catch!

Reply:
For `N` claims, this is `N` updates to the same file though, right?

If so, please open an issue to track this.

Reply:
Yes, it is - what should the issue contain, though, I'm not sure?

Reply:
Instead of doing `N` file accesses like:
```python
for claim in claims:
    claim.up_to_date(file)
```

we should do a single file access:

```python3
up_to_date(clams, file)
```

Reply:
We should fix this in this PR, I think.

Noting that there is a recursion happening in `_init_and_run_proof`, if this is of relevance.